server/sublist.go

```diff
@@ -83,7 +83,7 @@ func NewSublist() *Sublist {
 // Insert adds a subscription into the sublist
 func (s *Sublist) Insert(sub *subscription) error {
 	// copy the subject since we hold this and this might be part of a large byte slice.
-	subject := string(append([]byte(nil), sub.subject...))
+	subject := string(sub.subject)
```

Comment:
That is good information. I would still want to make sure we have a copy in the sub, even if we remove the L2 cache in he sublist, since this does acquire the write lock. Note the L1 is in the client readLoop.